code/Controllers/CMSMain.php

```diff
@@ -375,6 +375,16 @@ public function LinkListViewDeferred()
         return $this->Link('listview');
     }
 
+    /**
+     * Get the link for editing a page.
+     *
+     * @see CMSEditLinkExtension::getEditLinkForManagedDataObject()
+     */
+    public function getEditLinkForManagedDataObject(SiteTree $obj): string
+    {
+        return Controller::join_links(CMSPageEditController::singleton()->Link('show'), $obj->ID);
```

Comment:
```suggestion
        return $obj->CMSEditLink();
```
?

Reply:
Does this method even need to exist on CMSMain?  Is it here simply to make CMSMain more like a ModelAdmin?

Reply:
It's here so that the `CMSEditLinkExtension` on `SiteTree` can get its `CMSEditLink`.

Reply:
The recommended change will result in an unending recursive cycle. I can't make that change.